package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py

```diff
@@ -800,8 +878,17 @@ def count_by_type(self):
         d = self.u.atoms[self.hbonds[:, 1].astype(np.intp)]
         a = self.u.atoms[self.hbonds[:, 3].astype(np.intp)]
 
-        tmp_hbonds = np.array([d.resnames, d.types, a.resnames, a.types],
-                              dtype=str).T
+        if hasattr(d,"resnames"):
```

Comment:
I would probably say that this construct can be squashed, since they both stem from the same universe, if `d` has no resnames then `a` doesn't either. You can just do the one check?

Reply:
Oh good point!